au/io_test.cc

```diff
@@ -61,6 +61,8 @@ TEST(StreamingOutput, PrintsValueAndUnitLabel) {
     EXPECT_EQ(stream_to_string((feet / milli(second))(1.25)), "1.25 ft / ms");
 }
 
+TEST(StreamingOutput, PrintValueRepChar) { EXPECT_EQ(stream_to_string(feet(char{3})), "3 ft"); }
```

Comment:
What if we replace `char` with `int8_t`?

I think that would still be enough to reproduce the bug (https://godbolt.org/z/5cMoWGdWj), but it's probably better if we phrase it in terms of a quirk of the `int8_t` rep, so as not to suggest that people should be using `char` as a rep (which I think we don't want to encourage).

Reply:
That does introduce a dependency on `<cstdint>`, but probably not a bad idea.  I might be tempted to keep both cases (along with a comment about the `int8_t` resolving to a `char`)?

Reply:
Added in 71ed45e.